api/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the argoproj.io v1alpha1 API group
+// +kubebuilder:object:generate=true
+// +groupName=argoproj.io
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "argoproj.io", Version: "v1alpha1"}
```

Comment:
I am checking with Argo maintainers if it is ok to have the group defined as `argoproj.io` or if we should define it as `argoproj-labs.io`.
Once I get a confirmation from them, I'll reply back to you here.

Reply:
Hi @ggkhrmv, it is confirmed.
CRDs defined as part of `argoproj-labs` should not have the same domain/group from the ones defined by the `argoproj` org.

In your case the domain should be `argoproj-labs.io` and the Group should look like `rbac-operator.argoproj-labs.io`

In kubebuilder the domain is defined in the PROJECT file when the project is first initialized. Once that setup is done, running the following command would generate the CRD skeleton with the proper group:
`kubebuilder create api --group rbac-operator --version v1alpha1 --kind Role`

Reply:
will do, thank you :)

Reply:
@leoluz domain and group are changed in the latest commit :)